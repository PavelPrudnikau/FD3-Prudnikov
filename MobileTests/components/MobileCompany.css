.MobileCompany
{
  padding: 20px;
}

.MobileCompanyName
{
  font-weight: bold;
}

.MobileCompanyClients
{
  padding-top: 5px;
  border-collapse: collapse;
}
.MobileCompanyClients tr
{
  border-bottom: 1px solid black;
  border-top: 1px solid black;
}
.MobileCompanyClients td
{
  width: 120px;
  border-right: none; 
  border-left: none;
  padding:5px;
}

.MobileCompanyClients th
{
  text-align: left;
  padding:5px;
}

.button {

  padding: 7px 15px;
  text-align: center;
  font-size: 14px;
}

.addButton {

  padding: 7px 15px;
  text-align: center;
  font-size: 14px;
}

.EditButton {

  padding: 7px 15px;
  text-align: center;
  font-size: 14px;
}

.DeleteButton {

  padding: 7px 15px;
  text-align: center;
  font-size: 14px;
}



