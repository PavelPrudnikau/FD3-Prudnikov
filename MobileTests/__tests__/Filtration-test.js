"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import MobileCompany from '../components/MobileCompany';

test('работа Filtration', () => {

  let companyName='Velcom';
  let clientsArr=[ 
    {id:1, fam:"Иванов",    im:"Иван",      otch:"Иванович",    balance:200}, 
    {id:2, fam:"Сидоров",   im:"Сидор",     otch:"Сидорович",   balance:250}, 
    {id:3, fam:"Петров",    im:"Пётр",      otch:"Петрович",    balance:180},
    {id:4, fam:"Григорьев", im:"Григорий",  otch:"Григорьевич", balance:-220},
  ];
  // создаём тестовую версию компонента MobileCompany
  const componentCompany = renderer.create(
    <MobileCompany 
    name={companyName}
    clients={clientsArr}
  />
  );

  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentCompanyTree=componentCompany.toJSON();
  expect(componentCompanyTree).toMatchSnapshot();

  // найдём в вёрстке компонента кнопку "Активные"
  const buttonActive = componentCompany.root.find( el => el.type=='input' && el.props.value == 'Активные' ); 
  // и "нажмём" на неё
  buttonActive.props.onClick();

  // получаем уже изменённый снэпшот
  componentCompanyTree=componentCompany.toJSON();
  expect(componentCompanyTree).toMatchSnapshot();

  const buttonAll = componentCompany.root.find( el => el.type=='input' && el.props.value == 'Все' ); 
  // и "нажмём" на неё
  buttonAll.props.onClick();

  // получаем уже изменённый снэпшот
  componentCompanyTree=componentCompany.toJSON();
  expect(componentCompanyTree).toMatchSnapshot();

  // найдём в вёрстке компонента кнопку "Заблокированные"
  const buttonBlocked = componentCompany.root.find( el => el.type=='input' && el.props.value == 'Заблокированные' ); 
  // и "нажмём" на неё
  buttonBlocked.props.onClick();

  // получаем уже изменённый снэпшот
  componentCompanyTree=componentCompany.toJSON();
  expect(componentCompanyTree).toMatchSnapshot();
});
