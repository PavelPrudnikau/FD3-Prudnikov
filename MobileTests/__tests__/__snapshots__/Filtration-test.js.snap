// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа Filtration 1`] = `
<div
  className="MobileCompany"
>
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <br />
  <br />
  <table
    className="MobileCompanyClients"
  >
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="MobileClient"
        id={1}
      >
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="StatusActive"
        >
          active
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
        id={2}
      >
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="StatusActive"
        >
          active
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
        id={3}
      >
        <td>
          Петров
        </td>
        <td>
          Пётр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="StatusActive"
        >
          active
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
        id={4}
      >
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="StatusBlocked"
        >
          blocked
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <br />
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
  <br />
</div>
`;

exports[`работа Filtration 2`] = `
<div
  className="MobileCompany"
>
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <br />
  <br />
  <table
    className="MobileCompanyClients"
  >
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="MobileClient"
        id={1}
      >
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="StatusActive"
        >
          active
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
        id={2}
      >
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="StatusActive"
        >
          active
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
        id={3}
      >
        <td>
          Петров
        </td>
        <td>
          Пётр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="StatusActive"
        >
          active
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <br />
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
  <br />
</div>
`;

exports[`работа Filtration 3`] = `
<div
  className="MobileCompany"
>
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <br />
  <br />
  <table
    className="MobileCompanyClients"
  >
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="MobileClient"
        id={1}
      >
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="StatusActive"
        >
          active
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
        id={2}
      >
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="StatusActive"
        >
          active
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
        id={3}
      >
        <td>
          Петров
        </td>
        <td>
          Пётр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="StatusActive"
        >
          active
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
        id={4}
      >
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="StatusBlocked"
        >
          blocked
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <br />
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
  <br />
</div>
`;

exports[`работа Filtration 4`] = `
<div
  className="MobileCompany"
>
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <br />
  <br />
  <table
    className="MobileCompanyClients"
  >
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="MobileClient"
        id={4}
      >
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="StatusBlocked"
        >
          blocked
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <br />
  <input
    className="button"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
  <br />
</div>
`;
